include/finalfusion-cxx/Embeddings.hh

```diff
@@ -22,6 +27,15 @@ public:
    */
   Embeddings(std::string const &filename, bool mmap);
 
+  /**
+   * Constructor for embeddings in formats other than finalfusion.
+   *
+   * @param filename path to embeddings.
+   * @param format Format of embeddings
+   * @throws runtime_error if Embeddings could not be read.
+   */
+  Embeddings(std::string const &filename, NonFifuFormat const format);
```

Comment:
I'd strongly prefer static factory methods for the other embeddings per embedding type in the public API. This conveys more clearly what is going on and will make it possible to deprecate individual methods in the future, specify format-specific parameters (if necessary), etc.

Reply:
Sure, will do that later today.